src/solc_0.7/MultiGiveaway/ClaimERC721AndERC1155WithERC20.sol

```diff
@@ -0,0 +1,122 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.7.5;
+
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../common/Interfaces/IERC721Extended.sol";
+
+contract ClaimERC721AndERC1155WithERC20 {
+    bytes32 internal _merkleRoot;
+    IERC1155 internal immutable _asset;
+    IERC20 internal immutable _erc20Token;
```

Comment:
One thing that would be useful for our gems and catalyst is the ability to have multiple erc20
so instead of setting them in the constructor, their contract address would be part of the lead data

Reply:
and it would be an array of address and an array of amount